networking/src/p2p/binary_message.rs

```diff
@@ -161,4 +160,26 @@ impl<T: BinaryMessage> MessageHash for T {
         let bytes = self.as_bytes()?;
         Ok(blake2b::digest(&bytes))
     }
+}
+
+/// Trait for converting messages from/to HEX string
+pub trait Hexable {
+    /// Produce HEX string from the struct.
+    fn as_hex(&self) -> String;
```

Comment:
Use prefix `to_` because [naming guidelines](https://rust-lang-nursery.github.io/api-guidelines/naming.html)  prefix `as_` should be used for inexpensive methods.